lib/Listener/AddMissingIndicesListener.php

```diff
@@ -0,0 +1,19 @@
+<?php
+namespace OCA\Tables\Listener;
+
+use OCA\Tables\Service\Support\AuditLogServiceInterface;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\DB\Events\AddMissingIndicesEvent;
+final class AddMissingIndicesListener implements IEventListener {
+    public function __construct(protected AuditLogServiceInterface $auditLogService) {
+    }
+    public function handle(Event $event): void {
+        if (!$event instanceof AddMissingIndicesEvent) {
+            throw new \Exception('Error for testing only');
+            return;
+        }
+
+        $event->addMissingIndex('oc_tables_columns', 'table_id_index', ['table_id']);
```

Comment:
```suggestion
        $event->addMissingIndex('tables_columns', 'table_id_index', ['table_id']);
```

It fails because the table prefix (`oc_`) is added by the db abstraction.

Otherwise you may want to make the index name unique across tables (postgres requires that). A common pattern could be tables_columns_t_id. Shortening that makes sense to keep oracle support possible with 30 characters max https://docs.nextcloud.com/server/latest/developer_manual/basics/storage/database.html#supporting-more-databases